src/hidimstat/ensemble_clustered_inference.py

```diff
@@ -1,46 +1,314 @@
 import numpy as np
+from sklearn.base import clone
 from joblib import Parallel, delayed
 from sklearn.utils.validation import check_memory
 
-from .clustered_inference import clustered_inference
-from .multi_sample_split import aggregate_medians, aggregate_quantiles
+from hidimstat.desparsified_lasso import (
+    desparsified_lasso,
+    desparsified_lasso_pvalue,
+    desparsified_group_lasso_pvalue,
+)
+from hidimstat.utils import _subsampling, adaptive_quantile_aggregation
 
 
-def _ensembling(
-    list_beta_hat,
-    list_pval,
-    list_pval_corr,
-    list_one_minus_pval,
-    list_one_minus_pval_corr,
-    method="quantiles",
-    gamma_min=0.2,
+def _ungroup_beta(beta_hat, n_features, ward):
+    """
+    Ungroup cluster-level beta coefficients to individual feature-level
+    coefficients.
+
+    Parameters
+    ----------
+    beta_hat : ndarray, shape (n_clusters,) or (n_clusters, n_times)
+        Beta coefficients at cluster level
+    n_features : int
+        Number of features in original space
+    ward : AgglomerativeClustering
+        Fitted clustering object
```

Comment:
It should be possible to infer `n_features` from `ward` ...

Reply:
Sorry, I don't get what you mean. Ward is actually a special kind of  `FeatureAgglomeration` instance.

Reply:
I was wondering if there is a more general class than  `FeatureAgglomeration`.
My question comes from that I didn't find a Base class in scikitlearn for this type of object. 

If there is no Base class in this case, we can force the user to use this class and ask him to provide only the argument to configure it.

Reply:
No, `FeatureAgglomeration` is a general class. Indeed, the use will have to use and provide arguments.